.github/workflows/compile-macos-intel.yaml

```diff
@@ -0,0 +1,35 @@
+name: MacOS(ARM64)
+
+on: [pull_request]
+
+jobs:
+  compile-MacOS-ARM64:
+    runs-on: macos-latest
+    steps:
+    - name: Check Platform
+      run: |
+        if [ ! $(uname -m) = "arm64" ]; then
+            echo "Not on ARM64" >&2
+            exit 1
+        fi
+    - name: Setup MacOS
+      run: |
+        # Copied from mpi4py/mpi-publish
+        # create gfortran symlink
+        cd $(brew --prefix)/bin
+        gfortran=$(ls gfortran-* | sort | head -n 1)
+        sudo ln -s $gfortran gfortran
+        # install autotools
+        brew install autoconf
+        brew install automake
+        brew install libtool
+        # unlink libevent
+        brew unlink libevent || true
+    - uses: actions/checkout@v4
+      with:
+            submodules: recursive
+    - name: Build Open MPI
+      run: |
+        ./autogen.pl
+        ./configure
+        make
```

Comment:
Is there a reason to not `make -j`?  Might as well use all the CPUs to speed up the compilation.

Reply:
```suggestion
        make -j $(nproc)
```

Reply:
🤔 I have been seeing this for a few times
````
...
 make[2]: *** [libmpi_c_profile_la-scatter_init.lo] Error 1
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
make[2]: *** [libmpi_c_profile_la-free_mem.lo] Error 1
make[2]: *** [libmpi_c_profile_la-session_call_errhandler.lo] Error 1
make[2]: *** [libmpi_c_profile_la-op_create.lo] Error 1
make[2]: *** [libmpi_c_profile_la-irecv.lo] Error 1
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
clang: error: unable to execute command: posix_spawn failed: Resource temporarily unavailable
../../../libtool: fork: Resource temporarily unavailable
../../../libtool: fork: Resource temporarily unavailable
...
````

Reply:
That's fun.  IIRC, that runner has 3 CPUs.  Is $(nproc) actually returning 3?  If so, that's a weird error -- maybe try hard-coding to `make -j 2` (and putting a good comment explaining why)?